R/1.0_validate_reactive_teal_data.R

```diff
@@ -0,0 +1,79 @@
+validate_reactive_teal_data <- function(data) {
+  if (!is.reactive(data)) {
+    stop("The `teal_data_module` passed to `data` must return a reactive expression.", call. = FALSE)
+  }
+  reactive({
+    # custom module can return error
+    data_out <- tryCatch(data(), error = function(e) e)
+
+    # there is an empty reactive cycle on init!
+    if (inherits(data_out, "shiny.silent.error") && identical(data_out$message, "")) {
+      return(NULL)
+    }
+
+    # to handle qenv.error
+    if (inherits(data_out, "qenv.error")) {
+      validate(
+        need(
+          FALSE,
+          paste(
+            "Error when executing `teal_data_module` passed to `data`:\n ",
+            paste(data_out$message, collapse = "\n"),
+            "\n Check your inputs or contact app developer if error persists."
+          )
+        )
+      )
+    }
+
+    # to handle module non-qenv errors
+    if (inherits(data_out, "error")) {
+      validate(
+        need(
+          FALSE,
+          paste(
+            "Error when executing `teal_data_module` passed to `data`:\n ",
+            paste(data_out$message, collpase = "\n"),
+            "\n Check your inputs or contact app developer if error persists."
+          )
+        )
+      )
+    }
```

Comment:
```suggestion
    if (inherits(data_out, "qenv.error") || inherits(data_out, "error")) {
      validate(
        need(
          FALSE,
          paste(
            "Error when executing `teal_data_module` passed to `data`:\n ",
            paste(data_out$message, collapse = "\n"),
            "\n Check your inputs or contact app developer if error persists."
          )
        )
      )
    }
```

@gogonzo is it the same right now? Are you planning to make separate conditions for those two different error types? If not, we can merge into one condition

Reply:
I'll need to investigate this if I didn't write it wrong